### PR TITLE
add get article by id feature and update end point

### PR DESCRIPTION
**Pull request by karobarari**

### CODE REVIEW

**Review by nickkagame — CHANGES_REQUESTED** [review]
Hi Karo, Nice work on the happy path.  However we need some error handling for Status 400, invalid ID, e.g. string of "not-an-id"
Status 404, non existent ID, e.g. 0 or 9999.  
These need to be tested and have error handling middleware written for them.   Let me know if you have any questions!

File: __tests__/app.test.js [hunk 1]
```diff
@@ -58,3 +58,25 @@ describe("GET /api", () => {
...
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        });
+      });
+  });
```
Comment (nickkagame) [inline __tests__/app.test.js:81]: Still need error tests here!

**Review by nickkagame — CHANGES_REQUESTED** [review]
Great error handling!  but where are the tests?

File: __tests__/app.test.js (hunk 1 above)
Comment (karobarari) [reply __tests__/app.test.js:81]: ohh sorry!
I'll write them now

**Review by nickkagame — APPROVED** [review]
Great job!  Just remember to remove console logs before commits :-) Keep up the good work!

**Pull request merged**